### PR TITLE
Websocket security

### LINKED ISSUE #1326: Need encryption and authentication on websocket of the connecting user

### PR DESCRIPTION
**Pull request by sondreso**
**Issue**
Resolves #1326 

**Approach**
Create self-signed certificates and tokens. First commit will be dropped when https://github.com/equinor/libres/pull/1145 is merged.

### CODE REVIEW

**Comment by jondequinor** [comment]
We need to be able to run stuff insecurely too

**Comment by sondreso** [comment]
I know, it's on the agenda for when I'm going to fix the tests 😄

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -36,6 +36,11 @@ jobs:
       run: |
         pip install -r dev-requirements.txt
         pip install .[storage]
+        git clone --depth 1 -b websocket-security https://github.com/sondreso/libres.git tmp_libres
```
Comment (jondequinor) [inline .github/workflows/build.yml:39]: - [ ]

File: ert_shared/ensemble_evaluator/client.py [hunk 2]
```diff
@@ -14,15 +16,35 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
...
+        if url is None:
+            raise ValueError("url was None")
         self.url = url
+        self.token = token
+        self.cert = cert
+        self._extra_headers = Headers()
+        if token is not None:
+            self._extra_headers["token"] = token
+
+        self._cert = cert
```
Comment (jondequinor) [inline ert_shared/ensemble_evaluator/client.py:31]: Why two cert fields?

**Review by jondequinor — APPROVED** [review]
LGTM! =)

**Comment by sondreso** [comment]
Jenkins fails since bleeding has not been updated with newest libres version

**Pull request merged**